metricbeat/module/linux/memory/data.go

```diff
@@ -129,21 +136,28 @@ func getHugePages(hostfs resolve.Resolver) (common.MapStr, error) {
 }
 
 // GetVMStat gets linux vmstat metrics
-func GetVMStat() (*sysinfotypes.VMStatInfo, error) {
-	// TODO: We may want to pull this code out of go-sysinfo.
-	// It's platform specific, and not used by anything else.
-	h, err := sysinfo.Host()
+func GetVMStat(hostfs resolve.Resolver) (map[string]uint64, error) {
+	vmstat_file := hostfs.ResolveHostFS("proc/vmstat")
+	content, err := os.ReadFile(vmstat_file)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to read self process information")
-	}
-	vmstatHandle, ok := h.(sysinfotypes.VMStat)
-	if !ok {
-		return nil, errors.New("VMStat not available for platform")
+		return nil, errors.Wrapf(err, "error reading vmstat from %s", vmstat_file)
 	}
-	info, err := vmstatHandle.VMStat()
-	if err != nil {
-		return nil, errors.Wrap(err, "error getting VMStat info")
+
+	// I'm not a fan of throwing stuff directly to maps, but this is a huge amount of kernel/config specific metrics, and we're the only consumer of this for now.
+	vmstat := map[string]uint64{}
+	for _, metric := range strings.Split(string(content), "\n") {
```

Comment:
Nit: I think using `bufio.Scanner` is more idiomatic for reading file contents line by line.

Reply:
Really? I (personally) tend to lean towards stuff like `Split`, since passing around strings feels less opaque than a buffer.